Allow completion in strings and comments

yyoncho opened the pull request:
Some of the language servers do provide completion in strings and comments.

yyoncho committed:
Allow completion in strings and comments

tigersoldier commented on company-lsp.el:
```diff
@@ -451,7 +451,6 @@ See the documentation of `company-backends' for COMMAND and ARG."
     (prefix (and
              (bound-and-true-p lsp-mode)
              (lsp--capability "completionProvider")
-             (not (company-in-string-or-comment))
```
I'm afraid this has to be a per-server configuration. Not all language servers handle comments and strings well.

yyoncho commented:
@tigersoldier addressed your concerns - now the clients must explicitly declare that they want completion in comments. I have used nil as a default value for the newly introduced flag. Please, let me know what is your opinion on that.

yyoncho commented:
ping

yyoncho commented:
ping

yyoncho commented:
@tigersoldier is there anything blocking merging of this PR?

tigersoldier commented:
Sorry, I somehow filtered out notifications for this PR. Merged.
